vowpalwabbit/options.h

```diff
@@ -152,6 +153,24 @@ struct options_i
     return dynamic_cast<const typed_option<T>&>(base);
   }
 
+  // Check if option values exist and match.
+  // Add if it does not exist.
+  template<typename T>
+  bool add_or_check_options(const std::string& name, T expected_val){
+    if(was_supplied(name)) {
+      T found_val = get_typed_option<T>(name).value();
+      if(found_val != expected_val) {
+        return false;
+      }
+    }
+    else {
+      std::stringstream ss;
+      ss << expected_val;

```

Comment:
This would not work for vectors. Can you add a static assert to make sure we don't do that?

Reply:
Done.